### DISCUSSION TITLE
Client Variable Question

### DISCUSSION THREAD

**Discussion by wrldtvlr** [josephdadams/TallyArbiter#327] [Q&A]
[Created a new thread for these as the answer probably apply to all clients.]

In the M5 Matrix client there are two variables declared that I'm a bit confused about. 

listenerDeviceName is declared and initialized with a value. However, in the startup code, that value is overridden with another value that appends a serial number, thus creating a unique id. This value is passed up to the server in a couple of different places (wm.autoconnect, ArduinoOTA,setHostName and socket_Connected). Am I correct in thinking that the declared value is a placeholder and not really needed?

Somewhat similarly, in the Matrix code listenerType is declared and initialized with a value (in this case, M5). This value is not changed. It's just sent up to the server as a part of socketConnected. Presumably to let the server know how to talk back to the listener. However, when I looked at the M5StickC client code to confirm that it was using a similar or same value, it's not initialized there at all. Is this value being used in the server?

Thank you for tolerating the questions!

**Comment by josephdadams** [comment]
If I recall correctly, the listenerName “m5” was solely to display in the server GUI what type of client was connected. Since not all clients are created equally.

**Reply by wrldtvlr** [reply]
Thank you - yes, I see the listenerName (with device serial appended) displayed in the server web setup interface. Not sure what listenerType does, but I'll leave it alone.

**Reply by dougthebug** [reply]
I just figured this out. It's just DeviceName

tft.print(DeviceName);

Or whatever, will print the assigned text in Device Description, or the second entry for the device.
